contracts/secret-storage/SecretManager.sol

```diff
@@ -840,9 +842,11 @@ contract SecretManager is
 
         uint256 len = userStorage[_secretId].selectedEnclaves.length;
         address[] memory selectedStores;
+        uint8 env = userStorage[_secretId].env;
         for (uint256 index = 0; index < len; index++) {
-            if(userStorage[_secretId].selectedEnclaves[index].hasAcknowledgedStore)
-                selectedStores[index] = userStorage[_secretId].selectedEnclaves[index].enclaveAddress;
+            address enclaveAddress = userStorage[_secretId].selectedEnclaves[index].enclaveAddress;
+            if(userStorage[_secretId].selectedEnclaves[index].hasAcknowledgedStore && EXECUTORS.isNodePresentInTree(env, enclaveAddress))
```

Comment:
even if no executor is available for job, we can assign those jobs to enclaves not storing the given secret. This check of isNodePresentInTree is wrong.

Reply:
this function only gets those stores that are selected for the given secretId, and have the capacity to accept more jobs. If these stores aren't sufficient for the job, then extra enclaves are selected amongst the other executors only.

Reply:
but it is reverting in case all the assigned secret store have no jab capacity available.